src/garage/envs/dm_control/dm_control_env.py

```diff
@@ -8,33 +10,82 @@
 from garage.envs.dm_control.dm_control_viewer import DmControlViewer
 
 
-class DmControlEnv(gym.Env):
-    """
-    Binding for `dm_control <https://arxiv.org/pdf/1801.00690.pdf>`_
+def _flat_shape(observation):
+    """Returns the flattend shape of observation.
+
+    Args:
+        observation (np.ndarray): the observation
+
+    Returns:
+        np.ndarray: the flattened dimension of observation.
     """
+    return np.sum(int(np.prod(v.shape)) for k, v in observation.items())
+
+
+class DmControlEnv(gym.Env):
+    """Binding for `dm_control <https://arxiv.org/pdf/1801.00690.pdf>`."""
 
     def __init__(self, env, name=None):
+        """Create a DmControlEnv.
+
+        Args:
+            env (dm_control.suite.Task): The wrapped dm_control environment.
+            name (str): Name of the environment.
+        """
         self._name = name or type(env.task).__name__
         self._env = env
         self._viewer = None
 
     @classmethod
     def from_suite(cls, domain_name, task_name):
+        """Create a DmControl task given the domain name and task name.
+
+        Args:
+            domain_name (str): Domain name
+            task_name (str): Task name
+
+        Return:
+            dm_control.suit.Task: the dm_control task environment
+        """
         return cls(suite.load(domain_name, task_name),
                    name='{}.{}'.format(domain_name, task_name))
 
     def step(self, action):
+        """Step the environment.
+
+        Args:
+            action (object): input action
+
+        Returns:
+            Step: The time step after applying this action.
+        """
         time_step = self._env.step(action)
         return Step(
             flatten_observation(time_step.observation)['observations'],
             time_step.reward, time_step.step_type == StepType.LAST,
             **time_step.observation)
 
     def reset(self):
+        """Reset the environment.
+
+        Returns:
+            Step: The first time step.
+        """
         time_step = self._env.reset()
         return flatten_observation(time_step.observation)['observations']
```

Comment:
I think there needs to be a dict here too.

Reply:
I will rewrite DmControlEnv as well as other environments in a new PR. This file is in the commit because I once made experimental changes here. But it is basically the same as the one in master, with style/documentation fixes.